### PR TITLE
swr: display text on text input

### PR DESCRIPTION
**Pull request by FloVanGH**
For software renderer text is now displayed also for text input elements. Display of text cursor and selection will be part of a following PR.

### CODE REVIEW

**Review by ogoffart — COMMENTED** [review]
There could be a screenshot test with a bunch of TextInput with different properties set  (a bit like text.slint)

File: internal/core/software_renderer.rs [hunk 1]
```diff
@@ -1591,8 +1591,63 @@ impl<'a, T: ProcessScene> crate::item_rendering::ItemRenderer for SceneBuilder<'
...
+                    single_line: text_input.single_line(),
+                };
+
+                self.draw_text_paragraph(paragraph, physical_clip, offset, color);
+            }
+            #[cfg(feature = "software-renderer-systemfonts")]
+            fonts::Font::VectorFont(vf) => {
+                let paragraph = TextParagraphLayout {
+                    string: &string,
+                    layout: fonts::text_layout_for_font(&vf, &font_request, self.scale_factor),
```
Comment (ogoffart) [inline internal/core/software_renderer.rs:1638]: nitpick:  It looks like only this line is different from the two match arms.  perhaps worth to simplify this by only doing the layout in the match, and move the common code outside of the match.

**Review by ogoffart — APPROVED** [review]

**Comment by FloVanGH** [comment]
> There could be a screenshot test with a bunch of TextInput with different properties set (a bit like text.slint)

I thought about to implement the test after the whole thing is done with selection. What do you think?

File: internal/core/software_renderer.rs (hunk 1 above)
Comment (FloVanGH) [reply internal/core/software_renderer.rs:1638]: Yes but both layouts have incompatible types what would not work with the match.

**Pull request merged**